### ISSUE TITLE
[Bug]: Exception causes font to leak into default dalamud themes

### ISSUE THREAD

**Issue by Jaksuhn** [DelvUI/DelvUI#1140]
### What are you trying to do?

Nothing in Delv itself, this occurs "randomly". I've had it when combat starts and while fishing. I am not editing anything in Delv when this happens.

### What is the expected behavior?

Not to leak

### What actually happened?

Current dalamud font is overriden for any ImGui that uses it. 
![image](https://github.com/DelvUI/DelvUI/assets/9527380/ed4aa502-99f8-478b-af11-58f83610bfcf)

### Logs

```shell
18:06:32.889 | ERR Exception during raise of "Void Draw()"
System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.
   at DelvUI.Interface.Party.PartyFramesBar.ShouldDrawName(Character character, Boolean showingRaise, Boolean showingInvuln) in /work/repo/DelvUI/Interface/Party/PartyFramesBar.cs:line 643
   at DelvUI.Interface.Party.PartyFramesBar.GetElementsDrawActions(Vector2 origin) in /work/repo/DelvUI/Interface/Party/PartyFramesBar.cs:line 539
   at DelvUI.Interface.Party.PartyFramesHud.DrawChildren(Vector2 origin) in /work/repo/DelvUI/Interface/Party/PartyFramesHud.cs:line 404
   at DelvUI.Helpers.DraggablesHelper.DrawElements(Vector2 origin, HudHelper hudHelper, IList`1 elements, JobHud jobHud, DraggableHudElement selectedElement) in /work/repo/DelvUI/Helpers/DraggablesHelper.cs:line 101
   at DelvUI.Interface.HudManager.Draw(UInt32 jobId) in /work/repo/DelvUI/Interface/HudManager.cs:line 390
   at DelvUI.Plugin.Draw() in /work/repo/DelvUI/Plugin.cs:line 290
   at Dalamud.Utility.EventHandlerExtensions.HandleInvoke(Action act) in C:\goatsoft\companysecrets\dalamud\Utility\EventHandlerExtensions.cs:line 96
18:06:49.330 | ERR Exception during raise of "Void Draw()"
System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.
   at DelvUI.Interface.Party.PartyFramesBar.ShouldDrawName(Character character, Boolean showingRaise, Boolean showingInvuln) in /work/repo/DelvUI/Interface/Party/PartyFramesBar.cs:line 643
   at DelvUI.Interface.Party.PartyFramesBar.GetElementsDrawActions(Vector2 origin) in /work/repo/DelvUI/Interface/Party/PartyFramesBar.cs:line 539
   at DelvUI.Interface.Party.PartyFramesHud.DrawChildren(Vector2 origin) in /work/repo/DelvUI/Interface/Party/PartyFramesHud.cs:line 404
   at DelvUI.Helpers.DraggablesHelper.DrawElements(Vector2 origin, HudHelper hudHelper, IList`1 elements, JobHud jobHud, DraggableHudElement selectedElement) in /work/repo/DelvUI/Helpers/DraggablesHelper.cs:line 101
   at DelvUI.Interface.HudManager.Draw(UInt32 jobId) in /work/repo/DelvUI/Interface/HudManager.cs:line 390
   at DelvUI.Plugin.Draw() in /work/repo/DelvUI/Plugin.cs:line 290
   at Dalamud.Utility.EventHandlerExtensions.HandleInvoke(Action act) in C:\goatsoft\companysecrets\dalamud\Utility\EventHandlerExtensions.cs:line 96
18:07:17.011 | ERR Exception during raise of "Void Draw()"
System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.
   at DelvUI.Interface.Party.PartyFramesBar.ShouldDrawName(Character character, Boolean showingRaise, Boolean showingInvuln) in /work/repo/DelvUI/Interface/Party/PartyFramesBar.cs:line 643
   at DelvUI.Interface.Party.PartyFramesBar.GetElementsDrawActions(Vector2 origin) in /work/repo/DelvUI/Interface/Party/PartyFramesBar.cs:line 539
   at DelvUI.Interface.Party.PartyFramesHud.DrawChildren(Vector2 origin) in /work/repo/DelvUI/Interface/Party/PartyFramesHud.cs:line 404
   at DelvUI.Helpers.DraggablesHelper.DrawElements(Vector2 origin, HudHelper hudHelper, IList`1 elements, JobHud jobHud, DraggableHudElement selectedElement) in /work/repo/DelvUI/Helpers/DraggablesHelper.cs:line 101
   at DelvUI.Interface.HudManager.Draw(UInt32 jobId) in /work/repo/DelvUI/Interface/HudManager.cs:line 390
   at DelvUI.Plugin.Draw() in /work/repo/DelvUI/Plugin.cs:line 290
   at Dalamud.Utility.EventHandlerExtensions.HandleInvoke(Action act) in C:\goatsoft\companysecrets\dalamud\Utility\EventHandlerExtensions.cs:line 96
18:07:32.893 | ERR Exception during raise of "Void Draw()"
System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.
   at DelvUI.Interface.Party.PartyFramesBar.ShouldDrawName(Character character, Boolean showingRaise, Boolean showingInvuln) in /work/repo/DelvUI/Interface/Party/PartyFramesBar.cs:line 643
   at DelvUI.Interface.Party.PartyFramesBar.GetElementsDrawActions(Vector2 origin) in /work/repo/DelvUI/Interface/Party/PartyFramesBar.cs:line 539
   at DelvUI.Interface.Party.PartyFramesHud.DrawChildren(Vector2 origin) in /work/repo/DelvUI/Interface/Party/PartyFramesHud.cs:line 404
   at DelvUI.Helpers.DraggablesHelper.DrawElements(Vector2 origin, HudHelper hudHelper, IList`1 elements, JobHud jobHud, DraggableHudElement selectedElement) in /work/repo/DelvUI/Helpers/DraggablesHelper.cs:line 101
   at DelvUI.Interface.HudManager.Draw(UInt32 jobId) in /work/repo/DelvUI/Interface/HudManager.cs:line 390
   at DelvUI.Plugin.Draw() in /work/repo/DelvUI/Plugin.cs:line 290
   at Dalamud.Utility.EventHandlerExtensions.HandleInvoke(Action act) in C:\goatsoft\companysecrets\dalamud\Utility\EventHandlerExtensions.cs:line 96
```

### FFXIV Update

- [X] I have confirmed that I have the latest version of XIVLauncher and DelvUI.

**Comment by Zeffuro** [comment]
I accidently restarted the game when this happened to me so no log but this happened to me after someone left a P10S group and I exited the instance, the font blew up right then and there.

**Comment by Jaksuhn** [comment]
I can consistently get this to be reproduced when the dialog box for "do you want to accept this teleport" comes up. But it also happens in a variety of other areas, as I said previously while fishing and randomly when combat starts or ends.

Disabling and reenabling the plugin temporarily fixes it, until whatever is causing the leak happens again

**Comment by denoflionsx** [comment]
I'm seeing this exact same behavior but with a different stack trace when the problem occurs.

```06:08:27.429 | ERR Exception during raise of "Void Draw()"
System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.
   at Dalamud.Game.ClientState.Objects.Types.BattleChara.get_StatusList() in C:\goatsoft\companysecrets\dalamud\Game\ClientState\Objects\Types\BattleChara.cs:line 25
   at DelvUI.Helpers.Utils.GetTankInvulnerabilityID(BattleChara actor) in /work/repo/DelvUI/Helpers/Utils.cs:line 135
   at DelvUI.Interface.Party.PartyFramesBar.GetBarDrawActions(Vector2 origin, PluginConfigColor borderColor) in /work/repo/DelvUI/Interface/Party/PartyFramesBar.cs:line 263
   at DelvUI.Interface.Party.PartyFramesHud.DrawChildren(Vector2 origin) in /work/repo/DelvUI/Interface/Party/PartyFramesHud.cs:line 366
   at DelvUI.Helpers.DraggablesHelper.DrawElements(Vector2 origin, HudHelper hudHelper, IList`1 elements, JobHud jobHud, DraggableHudElement selectedElement) in /work/repo/DelvUI/Helpers/DraggablesHelper.cs:line 101
   at DelvUI.Interface.HudManager.Draw(UInt32 jobId) in /work/repo/DelvUI/Interface/HudManager.cs:line 390
   at DelvUI.Plugin.Draw() in /work/repo/DelvUI/Plugin.cs:line 290
   at Dalamud.Utility.EventHandlerExtensions.HandleInvoke(Action act) in C:\goatsoft\companysecrets\dalamud\Utility\EventHandlerExtensions.cs:line 96```

**Comment by Tischel** [comment]
Tentative fix in v1.7.1.2

**Comment by Jaksuhn** [comment]
Still happens. Slightly different stack trace. Also, I noticed any activity that introduces a lag spike in the game (switching to a different ROLE (not just job)) will reset the fonts.

```cs
10:40:11.934 | ERR Exception during raise of "Void Draw()"
System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.
   at Dalamud.Game.ClientState.Objects.Types.BattleChara.get_StatusList() in C:\goatsoft\companysecrets\dalamud\Game\ClientState\Objects\Types\BattleChara.cs:line 25
   at DelvUI.Interface.StatusEffects.StatusEffectsListHud.StatusEffectDataList(GameObject actor) in /work/repo/DelvUI/Interface/StatusEffects/StatusEffectsListHud.cs:line 148
   at DelvUI.Interface.StatusEffects.StatusEffectsListHud.StatusEffectsData() in /work/repo/DelvUI/Interface/StatusEffects/StatusEffectsListHud.cs:line 104
   at DelvUI.Interface.StatusEffects.StatusEffectsListHud.DrawChildren(Vector2 origin) in /work/repo/DelvUI/Interface/StatusEffects/StatusEffectsListHud.cs:line 284
   at DelvUI.Interface.Party.PartyFramesBar.<>c__DisplayClass27_0.<GetElementsDrawActions>b__0() in /work/repo/DelvUI/Interface/Party/PartyFramesBar.cs:line 513
   at DelvUI.Interface.HudElement.Draw(Vector2 origin) in /work/repo/DelvUI/Interface/HudElement.cs:line 41
   at DelvUI.Helpers.DraggablesHelper.Draw(Vector2 origin, HudHelper hudHelper, IList`1 elements, JobHud jobHud, DraggableHudElement selectedElement) in /work/repo/DelvUI/Helpers/DraggablesHelper.cs:line 176
   at DelvUI.Helpers.DraggablesHelper.DrawElements(Vector2 origin, HudHelper hudHelper, IList`1 elements, JobHud jobHud, DraggableHudElement selectedElement) in /work/repo/DelvUI/Helpers/DraggablesHelper.cs:line 132
   at DelvUI.Interface.HudManager.Draw(UInt32 jobId) in /work/repo/DelvUI/Interface/HudManager.cs:line 390
   at DelvUI.Plugin.Draw() in /work/repo/DelvUI/Plugin.cs:line 290
   at Dalamud.Utility.EventHandlerExtensions.HandleInvoke(Action act) in C:\goatsoft\companysecrets\dalamud\Utility\EventHandlerExtensions.cs:line 96
```

**Comment by JerryWester** [comment]
Consistently able to reproduce by having a party member leave the area.

```cs
ERR Exception during raise of "Void Draw()"
System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.
   at DelvUI.Interface.GeneralElements.CastbarHud.UpdateCurrentCast(Single& currentCastTime, Single& totalCastTime) in /work/repo/DelvUI/Interface/GeneralElements/CastbarHud.cs:line 162
   at DelvUI.Interface.GeneralElements.CastbarHud.DrawChildren(Vector2 origin) in /work/repo/DelvUI/Interface/GeneralElements/CastbarHud.cs:line 57
   at DelvUI.Interface.Party.PartyFramesBar.<>c__DisplayClass27_0.<GetElementsDrawActions>b__2() in /work/repo/DelvUI/Interface/Party/PartyFramesBar.cs:line 532
   at DelvUI.Interface.HudElement.Draw(Vector2 origin) in /work/repo/DelvUI/Interface/HudElement.cs:line 41
   at DelvUI.Helpers.DraggablesHelper.Draw(Vector2 origin, HudHelper hudHelper, IList`1 elements, JobHud jobHud, DraggableHudElement selectedElement) in /work/repo/DelvUI/Helpers/DraggablesHelper.cs:line 176
   at DelvUI.Helpers.DraggablesHelper.DrawElements(Vector2 origin, HudHelper hudHelper, IList`1 elements, JobHud jobHud, DraggableHudElement selectedElement) in /work/repo/DelvUI/Helpers/DraggablesHelper.cs:line 132
   at DelvUI.Interface.HudManager.Draw(UInt32 jobId) in /work/repo/DelvUI/Interface/HudManager.cs:line 390
   at DelvUI.Plugin.Draw() in /work/repo/DelvUI/Plugin.cs:line 290
   at Dalamud.Utility.EventHandlerExtensions.HandleInvoke(Action act) in C:\goatsoft\companysecrets\dalamud\Utility\EventHandlerExtensions.cs:line 96
```

**Comment by Tischel** [comment]
I appreciate the repro steps.
Another tentative fix in v1.7.1.3

**Comment by Jaksuhn** [comment]
New fix isn't out yet so I can't confirm yet but here's another reproducable way: when you become mounted

```cs
13:04:10.962 | ERR Exception during raise of "Void Draw()"
System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.
   at Dalamud.Game.ClientState.Objects.Types.BattleChara.get_StatusList() in C:\goatsoft\companysecrets\dalamud\Game\ClientState\Objects\Types\BattleChara.cs:line 25
   at DelvUI.Interface.StatusEffects.StatusEffectsListHud.StatusEffectDataList(GameObject actor) in /work/repo/DelvUI/Interface/StatusEffects/StatusEffectsListHud.cs:line 148
   at DelvUI.Interface.StatusEffects.StatusEffectsListHud.StatusEffectsData() in /work/repo/DelvUI/Interface/StatusEffects/StatusEffectsListHud.cs:line 104
   at DelvUI.Interface.StatusEffects.StatusEffectsListHud.DrawChildren(Vector2 origin) in /work/repo/DelvUI/Interface/StatusEffects/StatusEffectsListHud.cs:line 284
   at DelvUI.Interface.Party.PartyFramesBar.<>c__DisplayClass27_0.<GetElementsDrawActions>b__0() in /work/repo/DelvUI/Interface/Party/PartyFramesBar.cs:line 513
   at DelvUI.Interface.HudElement.Draw(Vector2 origin) in /work/repo/DelvUI/Interface/HudElement.cs:line 41
   at DelvUI.Helpers.DraggablesHelper.Draw(Vector2 origin, HudHelper hudHelper, IList`1 elements, JobHud jobHud, DraggableHudElement selectedElement) in /work/repo/DelvUI/Helpers/DraggablesHelper.cs:line 176
   at DelvUI.Helpers.DraggablesHelper.DrawElements(Vector2 origin, HudHelper hudHelper, IList`1 elements, JobHud jobHud, DraggableHudElement selectedElement) in /work/repo/DelvUI/Helpers/DraggablesHelper.cs:line 132
   at DelvUI.Interface.HudManager.Draw(UInt32 jobId) in /work/repo/DelvUI/Interface/HudManager.cs:line 390
   at DelvUI.Plugin.Draw() in /work/repo/DelvUI/Plugin.cs:line 290
   at Dalamud.Utility.EventHandlerExtensions.HandleInvoke(Action act) in C:\goatsoft\companysecrets\dalamud\Utility\EventHandlerExtensions.cs:line 96
```

**Comment by Tischel** [comment]
New fix in v1.7.1.4.
